packages/utilities/exposure-scanning/patch/index.test.js

```diff
@@ -0,0 +1,164 @@
+const { Headers, Response } = jest.requireActual('node-fetch');
+
+const mockHeaders = new Headers({ 'Content-Type': 'application/json' });
+
+const ResponseInit = {
+  url: 'hello',
+  headers: mockHeaders,
+  status: 200,
+};
+
+const mockRes = new Response(
+  JSON.stringify({
+    test: 'test',
+  }),
+  ResponseInit,
+);
+
+const mockFetch = jest.fn();
+jest.mock('node-fetch', () => ({
+  default: async (url, params) => {
+    mockFetch(url, JSON.parse(params.body).body);
+    return mockRes.clone();
+  },
+}));
+
+const { patchPr, patchIssue } = require('.');
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('Patching', () => {
+  it('should call fetch 5 times for the body, comments and review comments endpoints', async () => {
+    const reqBody = {
+      owner: 'bbc',
+      repo: 'psamemad',
+      id: '1234',
+    };
+
+    const pr = {
+      body: 'PR *',
+      comments: [
+        { id: 1, body: 'Added *' },
+        { id: 2, body: 'Nice work, LGTM!' },
+      ],
+      reviewComments: [
+        { id: 3, body: 'Smashed it!' },
+        { id: 4, body: 'Should there be * * for this?' },
+      ],
+    };
+
+    await patchPr(reqBody, pr);
+
+    expect(mockFetch).toHaveBeenCalledTimes(5);
+
+    expect(mockFetch).toHaveBeenCalledWith(
+      'https://api.github.com/repos/bbc/psamemad/pulls/1234',
+      pr.body,
+    );
```

Comment:
Is it worth checking the request method here to ensure it is `PATCH`: https://developer.mozilla.org/en-US/docs/Web/API/Request/method This would make sure the request actually makes a change to the PR body?

Reply:
Good idea 👍